db/migrate/20211018204729_change_grader_permissions_foreign_key.rb

```diff
@@ -0,0 +1,6 @@
+class ChangeGraderPermissionsForeignKey < ActiveRecord::Migration[6.1]
+  def change
+    add_reference :grader_permissions, :role, foreign_key:true
+    remove_column :grader_permissions, :users_id, type: :integer
```

Comment:
use `remove_reference` so that you can remove the foreign key as well at the same time